src/contrib/mlir/lowerer.cpp

```diff
@@ -538,6 +552,76 @@ namespace
 
 #undef REWRITER
 
+    template <typename OP>
+    void compute_binary_elementwise(Operation* op,
+                                    ArrayRef<Value*> operands,
+                                    PatternRewriter& rewriter,
+                                    DialectLoweringPass& m_pass)
+    {
+        auto loc = cast<OP>(op).getLoc();
+        auto result = m_pass.buildOutputDefs(op, rewriter)[0];
+        NGRAPH_CHECK(result->getType().isa<MemRefType>());
+        // Note that builder's current function is still the original function body.
+        // use getBlock to get the new block instead.
+
+        // get new operands
+        Value* lhs = operands[0];
+        Value* rhs = operands[1];
+
+        ScopedContext scope(rewriter, loc);
+        // Views
+        MemRefView vRes(result), vLHS(lhs), vRHS(rhs);
+        // Index Values
+        IndexedValue iRes(result), iLHS(lhs), iRHS(rhs);
+        // Bounds Index Handles
+        auto lbs = vLHS.getLbs();
+        auto ubs = vLHS.getUbs();
+        // Loop induction vars
+        auto ivs = IndexHandle::makeIndexHandles(vLHS.rank());
+        auto pivs = IndexHandle::makeIndexHandlePointers(ivs);
+        // Steps
+        auto steps = vLHS.getSteps();
+        // clang-format off
+        LoopNestBuilder(pivs, lbs, ubs, steps)(
+            // single stmt body
+            [&] {
+                    if (isa<NGAddOp>(op))
+                    {
+                        iRes(ivs) = iLHS(ivs) + iRHS(ivs);
+                    }
+                    else if (isa<NGSubOp>(op))
+                    {
+                        iRes(ivs) = iLHS(ivs) - iRHS(ivs);
+                    }
+                    else if (isa<NGMulOp>(op))
+                    {
+                        iRes(ivs) = iLHS(ivs) * iRHS(ivs);
+                    }
+                    else if (isa<NGDivOp>(op))
+                    {
+                        iRes(ivs) = iLHS(ivs) / iRHS(ivs);
+                    }
+                    else if (isa<NGGreaterOp>(op))
+                    {
+                        iRes(ivs) = ValueHandle(iLHS(ivs)) > ValueHandle(iRHS(ivs));
+                    }
+                    else if (isa<NGLessOp>(op))
+                    {
+                        iRes(ivs) = ValueHandle(iLHS(ivs)) < ValueHandle(iRHS(ivs));
+                    }
+                    else if (isa<NGMaxOp>(op))
+                    {
+                        iRes(ivs) = edsc::intrinsics::select(ValueHandle(iLHS(ivs)) > ValueHandle(iRHS(ivs)), ValueHandle(iLHS(ivs)) , ValueHandle(iRHS(ivs)));
+                    }
+                    else if (isa<NGMinOp>(op))
+                    {
+                        iRes(ivs) = edsc::intrinsics::select(ValueHandle(iLHS(ivs)) < ValueHandle(iRHS(ivs)), ValueHandle(iLHS(ivs)), ValueHandle(iRHS(ivs)));
+                    }
```

Comment:
We should be able to implement this in a more visitor pattern fashion instead of using a cascade of if-elses. That would be more efficient since when the template parameter is instantiated, we already know the specific op so we wouldn't need to execute `isa` again.